### PR TITLE
Reformat Docs Structure, Index page, and Getting Started Guide

### PR DESCRIPTION
**Pull request by smcguire-cmu**
- Adds buttons on index page to navigate to other sections.
- Restructures tutorials and example notebooks to separate sections, and combines installation and quickstart
- Rewords some of the Quickstart section

Docs for PR here: https://lsdb--406.org.readthedocs.build/en/406/

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -2,16 +2,17 @@
...
-particular those brought up by `LSST <https://www.lsst.org/about>`_.
+LSDB (Large Survey DataBase) is a python framework that enables simple, fast spatial analysis of extremely
+large astronomical catalogs (e.g. querying and crossmatching O(1B) sources). It aims to address large-scale
+data processing challenges, in particular those brought up by `LSST <https://www.lsst.org/about>`_.
 
-Built on top of Dask to efficiently scale and parallelize operations across multiple workers, it leverages
-the `HiPSCat <https://hipscat.readthedocs.io/en/stable/>`_ data format for surveys in a partitioned HEALPix
-(Hierarchical Equal Area isoLatitude Pixelization) structure.
+Built on top of Dask to efficiently scale and parallelize operations across multiple distributed workers, it
+uses the `HiPSCat <https://hipscat.readthedocs.io/en/stable/>`_ data format to effeciently perform spatial
```
Comment (camposandro) [inline docs/index.rst:14]:
```suggestion
uses the `HiPSCat <https://hipscat.readthedocs.io/en/stable/>`_ data format to efficiently perform spatial
```

**Review by camposandro — APPROVED** [review]
Great revamp overall! I really like the improvements on the Getting Started section.

File: docs/getting-started.rst [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+    .. code-block:: bash
+
+        conda create -n lsdb_env python=3.11
+        conda activate lsdb_env
+
+    We recommend Python versions **>=3.9, <=3.12**.
+
+LSDB can also be installed from source on `GitHub <https://github.com/astronomy-commons/lsdb>`_. See our
+advanced installation instructions in the :doc:`contributing guide </developer/contributing>`
```
Comment (delucchi-cmu) [inline docs/getting-started.rst:30]:
```suggestion
advanced installation instructions in the :doc:`contribution guide </developer/contributing>`.
```

File: docs/developer/contributing.rst [hunk 3]
```diff
@@ -1,6 +1,38 @@
...
+
+To install the package and a minimum number of dependencies you can run:
+
+.. code-block:: bash
+
+    python -m pip install .
+    python -m pip install pytest # to validate package installation
+
+In alternative, you can execute the `setup_dev` script which installs all the additional requirements
+to setup a development environment. Read more about contributing to LSDB in our
```
Comment (delucchi-cmu) [inline docs/developer/contributing.rst:22]: Remove sentence. This is that contribution guide.

File: docs/index.rst [hunk 4]
```diff
@@ -34,21 +64,22 @@ Learn more about contributing to this repository in our :doc:`Contribution Guide
 
    Home page <self>
    Installation <installation>
-   Getting Started <tutorials/quickstart>
-   Tutorials <tutorials>
```
Comment (delucchi-cmu) [inline docs/index.rst:38]: We had collapsed Tutorials and Example Notebooks into a single section intentionally. What's the reason to split them back out?

File: docs/index.rst [hunk 5]
```diff
@@ -34,21 +64,22 @@ Learn more about contributing to this repository in our :doc:`Contribution Guide
...
    :hidden:
-   :caption: Developer
 
-   API Reference <autoapi/index>
-   Contribution Guide <developer/contributing>
+   Contact us <contact>
 
 .. toctree::
    :hidden:
+   :caption: Developer
```
Comment (delucchi-cmu) [inline docs/index.rst:80]:
```suggestion
   :caption: Developers
```

File: docs/tutorials.rst [hunk 6]
```diff
@@ -1,10 +1,10 @@
-Tutorials
+Tutorial
```
Comment (delucchi-cmu) [inline docs/tutorials.rst:1]:
```suggestion
Tutorials
```

**Review by delucchi-cmu — APPROVED** [review]

**Review by camposandro — APPROVED** [review]

**Pull request merged**